src/dmd/dtoh.d

```diff
@@ -0,0 +1,1724 @@
+                {
+                    buf.printf("    %s(", sd.ident.toChars());
+                    bool first = true;
+                    foreach (m; *sd.members)
+                    {
+                        if (auto vd = m.isVarDeclaration())
+                        {
+                            if (!memberField(vd))
+                                continue;
+                            if (first)
+                                first = false;
+                            else
+                                buf.writestring(", ");
+                            assert(vd.type);
+                            assert(vd.ident);
+                            typeToBuffer(vd.type, vd.ident);
+                        }
+                    }
+                    buf.printf(") {");
+                    foreach (m; *sd.members)
+                    {
+                        if (auto vd = m.isVarDeclaration())
+                        {
+                            if (!memberField(vd))
+                                continue;
+                            buf.printf(" this->%s = %s;", vd.ident.toChars(), vd.ident.toChars());
+                        }
+                    }
+                    buf.printf(" }\n");
+                }
+            }
+            buf.writestring("};\n");
+
+            popAlignToBuffer(sd.alignment);
+            buf.writestring("\n");
+
+            auto savex = buf;
+            buf = checkbuf;
+            buf.writestring("    assert(sizeof(");
+            buf.writestring(sd.ident.toChars());
+            buf.printf(") == %d);\n", sd.size(Loc.initial));
+            buf = savex;
+        }
+        else
+        {
+            buf.writestring(";\n\n");
+        }
+    }
+
+    private void pushAlignToBuffer(uint alignment)
+    {
+        // DMD ensures alignment is a power of two
+        //assert(alignment > 0 && ((alignment & (alignment - 1)) == 0),
+        //       "Invalid alignment size");
+
+        // When no alignment is specified, `uint.max` is the default
+        if (alignment != uint.max)
+        {
+            buf.writestring("\n#if defined(__GNUC__) || defined(__clang__)\n");
+            // The equivalent of `#pragma pack(push, n)` is `__attribute__((packed, aligned(n)))`
+            // NOTE: removing the packed attribute will might change the resulting size
+            buf.printf("    __attribute__((packed, aligned(%d)))\n", alignment);
+            buf.writestring("#elif defined(_MSC_VER)\n");
+            buf.printf("    __declspec(align(%d))\n", alignment);
+            buf.writestring("#elif defined(__DMC__)\n");
+            buf.printf("    #pragma pack(push, %d)\n", alignment);
+            //buf.printf("#pragma pack(%d)\n", alignment);
+            buf.writestring("#endif\n");
+        }
+        else
+        {
+            buf.writeByte(' ');
+        }
+    }
+
+    private void popAlignToBuffer(uint alignment)
+    {
+        if (alignment != uint.max)
```

Comment:
```
if (alignment == uint.max)
    return;
...
```

Reply:
This was marked resolved but not addressed.

Reply:
Fixed